### DISCUSSION TITLE
Chat History appears always "New Chat"

### DISCUSSION THREAD

**Discussion by jpquinta** [open-webui/open-webui#2845] [General]
Hi, Since some new versions have been released, from a moment on, the history in the side menu always has the message "New Chat" and not a bit of the question or topic that was started appears.

## Environment

- **Open WebUI Version:** [0.2.4]
- **Ollama (if applicable):** [0.1.41]

**Confirmation:**
- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots
![2024-06-05_09h38_24](https://github.com/open-webui/open-webui/assets/42496083/c41d3812-e8d1-4377-abe7-02b22cfd1ab2)
![2024-06-05_09h45_10](https://github.com/open-webui/open-webui/assets/42496083/77ea110f-5482-4f44-b1a1-6779c042ce53)

Thank you
JP

**Comment by Fusseldieb** [comment]
Did you select a title generation model in your settings? Sometimes it defaults to an invalid or inexistent one.

**Comment by jpquinta** [comment]
Hi yes have that option activated

<img width="459" alt="2024-06-05_13h43_53" src="https://github.com/open-webui/open-webui/assets/42496083/62e46e1d-3fd8-45c9-a4e5-bc20ccde856b">

**Comment by jpquinta** [comment]
I have discovered that this option is in oposite action. 
That means when is "ON", is not create the Title on history, if you put in "Off", now it show us the title on History chat.....

**Comment by Fusseldieb** [comment]
Might have been an invalid entry which was corrected when you pressed Save.

![image](https://github.com/open-webui/open-webui/assets/4715129/867010b0-d700-4112-ac53-072a3a6735c7)

Usually you need to select something here.

**Comment by jpquinta** [comment]
Hi again,

Here goes more two screenshots of the settings and the behavior that I get without apply a fixed "local Model" or "External Models":
<img width="576" alt="2024-06-05_16h57_07" src="https://github.com/open-webui/open-webui/assets/42496083/1fcc5b6e-5ed9-4a9d-99b5-d807fda6bcff">

<img width="316" alt="2024-06-05_16h55_10" src="https://github.com/open-webui/open-webui/assets/42496083/819e8df6-eb0c-4cbe-9158-191d6f2e1563">

**Reply by Fusseldieb** [reply]
That's indeed strange. I'm using 2.4, too, and mine's "ON", and works as expected.

**Reply by p-try** [reply]
I am experiencing the same issue and I can add the following (hopefully useful) information:

For me, this issue only appears when I connect to an external Ollama instance. My Open WebUI is hosted on my own machine, but I frequently connect to an externally hosted Ollama instance (mostly on RunPod.io) for GPU acceleration. In this case, automatic title generation does not work, unless I set the auto-title generation to OFF. Then it does!

Whenever I run the prompts on my local Ollama (host.docker.internal), the opposite applies: Automatic title generation works when it is set to ON.

**Reply by Seantourage** [reply]
I can confirm and that is really strange. An interesting note is when i first set it up on v0.3.1 I could swear it did create a title on my first chat but then stopped from that point. I updated to v0.3.2, reinstalled the docker container, deleted the volume and still couldn't get it to work again.

Once I tried the above suggestion of disabling it in my use profile it now works. I'm using an external ollama instance.

**Reply by justinh-rahb** [reply]
Thank you for bringing this to our attention. If the title generation task fails, the chat will be renamed as 'New Chat'. However, if it's disabled, it will simply replace it with the prompt.

**Reply by Seantourage** [reply]
That makes sense. I did some digging and it appears to be the cloudflare tunnel I'm using for WebUI to access my ollama server. When I changed it the ollama server address to the IP and Port it worked.

On the browser I get a streamed response to my query but the console shows this error:

```
Failed to load resource: net::ERR_CONTENT_DECODING_FAILED
index.ts:194 TypeError: Failed to fetch
    at index.ts:191:15
    at async $ (index.ts:176:14)
    at async W (Chat.svelte:1187:24)
    at async nt (Chat.svelte:823:10)
    at async Chat.svelte:508:7
    at async Promise.all (index 0)
    at async F (Chat.svelte:450:52)
    at async lt (Chat.svelte:374:23)
```

While everything looks fine in the docker console:

```
INFO:apps.ollama.main:url: http://ollama.redacteddomain.com
INFO:     67.160.***.***:0 - "POST /ollama/api/chat HTTP/1.1" 200 OK
INFO:     127.0.0.1:49486 - "GET /health HTTP/1.1" 200 OK
llama3-Q6:latest
INFO:     67.160.***.***:0 - "POST /api/chat/completed HTTP/1.1" 200 OK
INFO:     67.160.***.***:0 - "POST /api/v1/chats/b31a9195-9af0-4a22-9d7e-81a0b01e6e61 HTTP/1.1" 200 OK
INFO:     67.160.***.***:0 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
generate_title
llama3-Q6:latest
INFO:apps.ollama.main:url: http://ollama.redacteddomain.com
INFO:     67.160.***.***:0 - "POST /api/task/title/completions HTTP/1.1" 200 OK
INFO:     67.160.***.***:0 - "POST /api/v1/chats/b31a9195-9af0-4a22-9d7e-81a0b01e6e61 HTTP/1.1" 200 OK
INFO:     67.160.***.***:0 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
INFO:     67.160.***.***:0 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
['tklJ-Li2JKo_QI0-AAAF']
```

It's a unique case so don't expect a fix, but thought I'd follow up with additional information to keep everyone informed.
